wasm/main.py

```diff
@@ -438,7 +443,7 @@ def spec_validate_call_indirect(C, x):
     if C["tables"] == None or len(C["tables"]) < 1:
         raise InvalidModule("invalid")
     limits, elemtype = C["tables"][0]
-    if elemtype != "anyfunc":
+    if elemtype != FuncType:
```

Comment:
Should this be `elemtype is not FuncType`?

Reply:
Actually, here's another thing: down in the "Binary Format" section of the w3c spec, it appears that types are actually encoded into modules as metadata for validity checking.  According to that section of the spec, the `funcref` elemtype has a specific encoding of `0x70` (https://webassembly.github.io/spec/core/bikeshed/index.html#table-types%E2%91%A4).  Therefore, it actually seems appropriate to me that `elemtype` should refer to a singleton value called something like `FuncRef` and not `FuncType`.  In the same section, function types have their own separate encodings based on the vectors of input and output types (https://webassembly.github.io/spec/core/bikeshed/index.html#function-types%E2%91%A4).

Reply:
Another way to put it is that the concepts of `funcref` and the type construct of "function types", which might be referred to by a label such as `FuncType`, seem to be regarded differently in terms of the spec even though the spec says that `funcref` is meant to represent the union of function types.  The `funcref` label still seems to be used fairly often and corresponds more or less with a particular piece of encoded metadata.

Reply:
I'm convinced.